testsuite/tests/lib-unix/common/wait_nohang.ml

```diff
@@ -15,3 +15,19 @@ let () =
   match Sys.remove "plop" with
   | () ->  print_endline "OK"
   | exception (Sys_error _) -> print_endline "ERROR"
+
+let () =
+  let fd = Unix.openfile "plip" [O_CREAT; O_WRONLY] 0o666 in
+  let pid =
+    Unix.create_process "sleep" [|"sleep"; "0.01"|] Unix.stdin fd Unix.stderr
+  in
+  Unix.close fd;
+  let pid, status = Unix.waitpid [WNOHANG] pid in
+  assert (pid = 0);
+  assert (status = WEXITED 0);
+  while fst (Unix.waitpid [WNOHANG] pid) = 0 do
```

Comment:
This test looks very fragile to me.  On a busy machine, the `sleep 0.01` could finish before the tester (the present OCaml code) gets to run the `Unix.waitpid`.  Could you please write a test that has no race conditions?  E.g.
* Create a pipe
* Run `cat` with stdin = the pipe and stdout = /dev/null
* Check that `Unix.waitpid [WNOHANG] pid` is pid = 0, status = WEXITED 0.
* Close the pipe, causing `cat` to terminate.
* Check again  `Unix.waitpid [WNOHANG] pid`  in a loop like in the current code.